clang/test/SemaSYCL/array-num-elements-exceeded.cpp

```diff
@@ -0,0 +1,17 @@
+// RUN: %clang_cc1 -I %S/Inputs -fsycl -fsycl-is-device -verify -fsyntax-only %s
+#include "big_constant.h"
+
+template <typename Name, typename Func>
+__attribute__((sycl_kernel)) void kernel_single_task(Func kernelFunc) {
+
+  kernelFunc(); // expected-note {{called by 'kernel_single_task<fake_kernel, (lambda}}
+}
+
+int main() {
+  double res;
+  kernel_single_task<class fake_kernel>([=]() mutable {
+    res = big[0]; // expected-error {{Global variable has a constant array with number of elements greater than OpConstantComposite can have (65532).Can the array be split?}}
```

Comment:
I'm not sure that this is good error message: in particular I'm not sure that reference to `OpConstantComposite` would be useful for end-user.

What about something like: "Due to SPIR-V intermediate format limitations, constant arrays with number of elements more then 65532 cannot be used. You can workaround this limitation by splitting the array into a several ones" ?

Reply:
Thank you for wordsmithing. This was a placeholder text borrowed from the SPIR_V PR. I forgot to mention that it was open for wordsmithing.

Reply:
Also, I added `mutable` to the lambda, since I wanted to assign `res` (captured by value) inside the lambda. 
I hope this is okay?